docs/__tests__/inline-filter.test.tsx

```diff
@@ -0,0 +1,135 @@
+import { render } from '@testing-library/react';
+import { InlineFilter } from '../src/components/InlineFilter';
+import { useRouter } from 'next/router';
+
+jest.mock('next/router', () => ({
+  useRouter: jest.fn(),
+}));
+// mock a return value on useRouter
+useRouter.mockReturnValue({
+  query: {
+    platform: 'react',
+  },
+});
+
+describe('InlineFilter', () => {
+  it('should display children of matching filters', () => {
+    const { container } = render(
+      <div>
+        <InlineFilter filters={['react']}>
+          <span id="react-content">React Content</span>
+        </InlineFilter>
+        <InlineFilter filters={['angular']}>
+          <span id="angular-content">Angular Content</span>
+        </InlineFilter>
+      </div>
+    );
+    const reactNode = container.querySelector('#react-content');
+    const angularNode = container.querySelector('#angular-content');
+    expect(reactNode).not.toBeNull();
+    expect(angularNode).toBeNull();
+  });
+
+  it('should use exact matches for filters', () => {
+    const { container } = render(
+      <div>
+        <InlineFilter filters={['react']}>
+          <span id="react-content">React Content</span>
+        </InlineFilter>
+        <InlineFilter filters={['React']}>
+          <span id="filter1">Content</span>
+        </InlineFilter>
+        <InlineFilter filters={['@react']}>
+          <span id="filter2">Content</span>
+        </InlineFilter>
+        <InlineFilter filters={['1react']}>
+          <span id="filter3">Content</span>
+        </InlineFilter>
+      </div>
+    );
+    const reactNode = container.querySelector('#react-content');
+    const filter1 = container.querySelector('#filter1');
+    const filter2 = container.querySelector('#filter2');
+    const filter3 = container.querySelector('#filter3');
+    expect(reactNode).not.toBeNull();
+    expect(filter1).toBeNull();
+    expect(filter2).toBeNull();
+    expect(filter3).toBeNull();
+  });
+
+  it('should always render the all filter', () => {
+    const { container } = render(
+      <div>
+        <InlineFilter filters={['react']}>
+          <span id="react-content">React Content</span>
+        </InlineFilter>
+        <InlineFilter filters={['all']}>
+          <span id="filter1">Content</span>
+        </InlineFilter>
+        <InlineFilter filters={['not-all']}>
+          <span id="filter2">Content</span>
+        </InlineFilter>
+      </div>
+    );
+    const reactNode = container.querySelector('#react-content');
+    const filter1 = container.querySelector('#filter1');
+    const filter2 = container.querySelector('#filter2');
+    expect(reactNode).not.toBeNull();
+    expect(filter1).not.toBeNull();
+    expect(filter2).toBeNull();
+  });
+
+  it('should not render anything if no or an invalid filter is passed in', () => {
```

Comment:
Rather than declaring multiple cases within the same test, each case be isolated using `it.each`, example:
```
it.each([undefined, true]).('renders nothing when provided with %s as the provided value of filters`, (filters) => {
  const { container } = render(
     <InlineFilter filters={filters}>
        <span id="filtered">Content</span>
      </InlineFilter>
  );

  const filtered = container.querySelector('#filtered');
  expect(filtered).toBeNull();
});
```